src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractShuffle.java

```diff
@@ -39,8 +39,8 @@ abstract class AbstractShuffle<E> extends VectorShuffle<E> {
     /*package-private*/
     abstract VectorPayloadMF indices();
 
-    static VectorPayloadMF prepare(int length, int[] indices, int offset) {
-        VectorPayloadMF payload = VectorPayloadMF.newInstanceFactory(byte.class, length);
+    static VectorPayloadMF prepare(Class<?> elemType, int length, int[] indices, int offset, boolean is_max_species) {
+        VectorPayloadMF payload = VectorPayloadMF.newShuffleInstanceFactory(elemType, length, is_max_species);
```

Comment:
Same as VectorMask. We can create a common method which returns a shuffe payload from the given species.